### PR TITLE
Add pod name and namespace info when inject sidecar error

### PR DESCRIPTION
**Pull request by yuzhiquan**
**Please provide a description of this PR:**
We have received some error logs about injecting sidecar, but can not figure out which pod should I check, we have thousands pod, and it's hard, so I want add pod name and namespace info when injecting sidecar error:
Now

```
2024-09-11T09:30:24.387340Z    error    Pod injection failed: failed to run injection template: failed applying injection overlay: unmarshal patched pod: quantities must match the regular expression '^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$'

```
 then
```bash
2024-09-11T09:30:24.387340Z    error    Pod injection failed: failed to run injection template: failed applying injection overlay: unmarshal patched pod(${pod_name} ${pod_namespace}): quantities must match the regular expression '^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$'
```

### CODE REVIEW

File: pkg/kube/inject/webhook.go [hunk 1]
```diff
@@ -1068,7 +1068,7 @@ func applyOverlayYAML(target *corev1.Pod, overlayYAML []byte) (*corev1.Pod, erro
 	}
 
 	if err := json.Unmarshal(patched, &pod); err != nil {
-		return nil, fmt.Errorf("unmarshal patched pod: %v", err)
+		return nil, fmt.Errorf("unmarshal patched pod(%s %s): %v", target.Name, target.Namespace, err)
```
Comment (hzxuzhonghu) [inline pkg/kube/inject/webhook.go:1071]:
```suggestion
		return nil, fmt.Errorf("unmarshal patched pod[%s/%s] failed: %v", target.Name, target.Namespace, err)
```

**Comment by wulianglongrd** [comment]
please use the `namespace/name` format.

/ok-to-test

**Comment by howardjohn** [comment]
nice, should we just put the info in the top level `Pod injection faile` though so we don't need to put it in every error? probably there are some errors we miss

**Comment by yuzhiquan** [comment]
> nice, should we just put the info in the top level `Pod injection faile` though so we don't need to put it in every error? probably there are some errors we miss

Cool, move it.

**Comment by yuzhiquan** [comment]
> please use the `namespace/name` format.
> 
> /ok-to-test

Thanks.

**Comment by wulianglongrd** [comment]
> > nice, should we just put the info in the top level `Pod injection faile` though so we don't need to put it in every error? probably there are some errors we miss
> 
> Cool, move it.

We can log with labels, there is an example:
https://github.com/istio/istio/blob/d65eb93abe73c8b79156e425a09c8e9b0fd5f633/cni/pkg/nodeagent/informers.go#L201

**Comment by yuzhiquan** [comment]
> > > nice, should we just put the info in the top level `Pod injection faile` though so we don't need to put it in every error? probably there are some errors we miss
> > 
> > 
> > Cool, move it.
> 
> We can log with labels, there is an example:
> 
> https://github.com/istio/istio/blob/d65eb93abe73c8b79156e425a09c8e9b0fd5f633/cni/pkg/nodeagent/informers.go#L201

That's a good idea, for this line, I did not find a proper way to achieve this target;
https://github.com/istio/istio/blob/09e09fcb82d70724709f7a7eb08ed9fc373c4662/pkg/kube/inject/webhook.go#L1341

File: pkg/kube/inject/webhook.go [hunk 2]
```diff
@@ -1174,7 +1174,7 @@ func (wh *Webhook) inject(ar *kube.AdmissionReview, path string) *kube.Admission
 
 	patchBytes, err := injectPod(params)
 	if err != nil {
-		handleError(fmt.Sprintf("Pod injection failed: %v", err))
+		handleError(fmt.Sprintf("Pod[%s/%s] injection failed: %v", podName, pod.Namespace, err))
```
Comment (wulianglongrd) [inline pkg/kube/inject/webhook.go:1177]:
```suggestion
		handleError(fmt.Sprintf("Pod %s/%s injection failed: %v", pod.Namespace, podName, err))
```

**Comment by wulianglongrd** [comment]
Sorry for the confusion. I think it can be merged first, and then I'll see if I can make some improvements.

**Comment by yuzhiquan** [comment]
> Sorry for the confusion. I think it can be merged first, and then I'll see if I can make some improvements.

Thanks.

**Review by hzxuzhonghu — APPROVED** [review]

**Review by howardjohn — COMMENTED** [review]
lgtm, thanks

**Pull request merged**